### PR TITLE
Refactor env conversion to eliminate `Value::follow_cell_path_not_from_user_input`

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Replaces the only usage of `Value::follow_cell_path_not_from_user_input` with some `Record::get`s.

# User-Facing Changes
Breaking change for `nu-protocol`, since `Value::follow_cell_path_not_from_user_input` was deleted.

Nushell now reports errors for when environment conversions are not closures.

### CODE REVIEW

File: crates/nu-engine/src/env.rs [hunk 1]
```diff
@@ -367,59 +367,48 @@ fn get_converted_value(
...
+                    ConversionResult::ConversionError(ShellError::MissingParameter {
+                        param_name: "block input".into(),
+                        span: from_span,
+                    })
                 }
-                _ => ConversionResult::CellPathError,
             }
-        } else {
-            ConversionResult::CellPathError
+            _ => ConversionResult::CellPathError,
```
Comment (IanManske) [inline crates/nu-engine/src/env.rs:411]: I think this should instead propagate the `ShellError` as `ConversionResult::ConversionError`. Also, maybe the `ShellError::CantConvert` should be transformed to `ShellError::UnsupportedConfigValue`.

Comment (sholderbach) [reply crates/nu-engine/src/env.rs:411]: Yeah feel free to map that to more appropriate errors as you see fit.

File: crates/nu-engine/src/env.rs [hunk 2]
```diff
@@ -367,59 +367,48 @@ fn get_converted_value(
...
-                        if let Some(var_id) = &var.var_id {
-                            stack.add_var(*var_id, orig_val.clone());
-                        }
+    let conversions = stack.get_env_var(engine_state, ENV_CONVERSIONS);
+    let conversion = conversions
+        .as_ref()
+        .and_then(|val| val.as_record().ok())
+        .and_then(|record| record.get(name))
+        .and_then(|val| val.as_record().ok())
+        .and_then(|record| record.get(direction));
```
Comment (sholderbach) [inline crates/nu-engine/src/env.rs:376]: Rust, you love to see it 😁

**Review by sholderbach — APPROVED** [review]
Looks fine to me

**Comment by IanManske** [comment]
Hmm... I'm not sure why but I'm getting some jarbbled error output if the conversion is not a closure.
```
/home/iamaError: nu::shell::cant_convert

                                          × Can't convert to closure.
                                                                         ╭─[/home/iama/.config/nushell/env.nu:23:1]
                                                                                                                    23 │     from_string: "nope"
                                                                                                                                                 24 │     to_string: "nope"
                                                                                                                                                                               ·                ───┬──
                                                                                                                                                                                                          ·                   ╰── can't convert string to closure
          25 │   }
                      ╰────

/home/iama
Error: nu::shell::cant_convert

  × Can't convert to closure.
    ╭─[/home/iama/.config/nushell/env.nu:23:1]
 23 │     from_string: "nope"
 24 │     to_string: "nope"
    ·                ───┬──
    ·                   ╰── can't convert string to closure
 25 │   }
    ╰────
```
And it only happens on the fourth time I press enter (I think once the text reaches the bottom of the terminal).

**Comment by sholderbach** [comment]
> And it only happens on the fourth time I press enter (I think once the text reaches the bottom of the terminal).
> 
> Should we treat this as a separate issue?

Can you reproduce something similar on `main`?
I don't know what the exact error reporting mechanism is here but that is probably a problem somewhere else.

To me it looks like it is outputting while the terminal is still in raw mode so the carriage return is not automatically added for `\n`.

I think worth filing a separate issue.

**Comment by IanManske** [comment]
Ok, opened a separate issue. That one was a doozy to figure out.

**Comment by sholderbach** [comment]
Let's go with this!

**Pull request merged**